cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -638,23 +667,38 @@ private SubmitJobRequest getSubmitJobRequest(RuntimeJobInfo runtimeJobInfo, List
       }
     }
 
+    Job.Builder dataprocJobBuilder = Job.newBuilder()
+      // use program run uuid as hadoop job id on dataproc
+      .setReference(JobReference.newBuilder().setJobId(getJobId(runInfo)))
+      // place the job on provisioned cluster
+      .setPlacement(JobPlacement.newBuilder().setClusterName(clusterName).build())
+      // add same labels as provisioned cluster
+      .putAllLabels(labels)
+      // Job label values must match the pattern '[\p{Ll}\p{Lo}\p{N}_-]{0,63}'
+      // Since program name and type are class names they should follow that pattern once we remove all
+      // capitals
+      .putLabels(LABEL_CDAP_PROGRAM, runInfo.getProgram().toLowerCase())
+      .putLabels(LABEL_CDAP_PROGRAM_TYPE, runInfo.getProgramType().toLowerCase())
+      .setHadoopJob(hadoopJobBuilder.build());
+
+    GetClusterRequest getClusterRequest = GetClusterRequest.newBuilder()
+      .setClusterName(clusterName).setProjectId(projectId).setRegion(region).build();
+
+    // if the dataproc cluster has driver pools enabled add the driver scheduling config
+    if (getClusterControllerClient().getCluster(getClusterRequest).getConfig().getAuxiliaryNodeGroupsCount() > 0) {
```

Comment:
For feature preview please add a flag / feature flag to be able to disable this. Some customers may use static clusters for different purposes, so we should be able to disable it if needed event on a cluster with driver pools

Reply:
I understand the point of disabling the feature with the help of flag but if a Cx is launching a pipeline on dataproc cluster with driver pools, the job submission on dataproc will fail if driver scheduling config is not present in it. So they will not be use it for any other purpose through CDAP unless you add it.

Do you still think we should disable it??

Reply:
Oh, in this case please disregard. I thought that for drvierpool-enabled cluster you can do both driverpool and regular jobs